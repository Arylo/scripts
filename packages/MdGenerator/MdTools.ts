const INDENT_SPACE_LENGTH = 2

export function indent (level: number, content = '') {
  return `${Array((level - 1) * INDENT_SPACE_LENGTH).fill(' ').join('')}${content}`
}

export function enter() {
  return '\n'
}

export type HeaderLevel = 1 | 2 | 3 | 4 | 5
const header = (level: HeaderLevel) => (text = '') => `${Array(level).fill('#').join('')} ${text}`.trim()

export function h1 (...args: Parameters<ReturnType<typeof header>>) {
  return header(1)(...args)
}
export function h2 (...args: Parameters<ReturnType<typeof header>>) {
  return header(2)(...args)
}
export function h3 (...args: Parameters<ReturnType<typeof header>>) {
  return header(3)(...args)
}
export function h4 (...args: Parameters<ReturnType<typeof header>>) {
  return header(4)(...args)
}
export function h5 (...args: Parameters<ReturnType<typeof header>>) {
  return header(5)(...args)
}

export function text(content = '') {
  return content
}
export function anchor(key: string, url: string) {
  return `[${key}]: ${url}`
}

export function hyperlink(label: string, url: string) {
  return `[${label}](${url})`
}

export function hyperlinkWithKey(label: string, key: string) {
  return `[${label}][${key}]`
}

export function image(url: string, alt = '') {
  return `!${hyperlink(alt, url)}`
}

export function imageByKey(key: string, alt = '') {
  return `!${hyperlinkWithKey(alt, key)}`
}

export function listItem(text = '') {
  return `- ${text}`.trimEnd()
}

export function taskItem(text = '', { selected = false } = {}) {
  return `[${selected ? 'x' : ' '}] ${text}`.trimEnd()
}
