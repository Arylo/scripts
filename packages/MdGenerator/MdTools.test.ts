import { test, expect, describe } from 'vitest'
import * as mdTools from './MdTools'

describe('Markdown Tools', () => {
  describe(mdTools.indent.name, () => {
    test('Level 1', () => expect(mdTools.indent(1)).toMatchSnapshot())
    test('Level 1 with text', () => expect(mdTools.indent(1, 'text')).toMatchSnapshot())
    test('Level 2', () => expect(mdTools.indent(2)).toMatchSnapshot())
    test('Level 2 with text', () => expect(mdTools.indent(2, 'text')).toMatchSnapshot())
    test('Level 3', () => expect(mdTools.indent(3)).toMatchSnapshot())
    test('Level 3 with text', () => expect(mdTools.indent(3, 'text')).toMatchSnapshot())
    test('Level 4', () => expect(mdTools.indent(4)).toMatchSnapshot())
    test('Level 4 with text', () => expect(mdTools.indent(4, 'text')).toMatchSnapshot())
    test('Level 5', () => expect(mdTools.indent(5)).toMatchSnapshot())
    test('Level 5 with text', () => expect(mdTools.indent(5, 'text')).toMatchSnapshot())
  })

  test(mdTools.enter.name, () => {
    expect(mdTools.enter()).toMatchSnapshot()
  })

  describe(mdTools.h1.name, () => {
    test('No title', () => expect(mdTools.h1()).toMatchSnapshot())
    test('Exist title', () => expect(mdTools.h1('Text')).toMatchSnapshot())
  })
  describe(mdTools.h2.name, () => {
    test('No title', () => expect(mdTools.h2()).toMatchSnapshot())
    test('Exist title', () => expect(mdTools.h2('Text')).toMatchSnapshot())
  })
  describe(mdTools.h3.name, () => {
    test('No title', () => expect(mdTools.h3()).toMatchSnapshot())
    test('Exist title', () => expect(mdTools.h3('Text')).toMatchSnapshot())
  })
  describe(mdTools.h4.name, () => {
    test('No title', () => expect(mdTools.h4()).toMatchSnapshot())
    test('Exist title', () => expect(mdTools.h4('Text')).toMatchSnapshot())
  })
  describe(mdTools.h5.name, () => {
    test('No title', () => expect(mdTools.h5()).toMatchSnapshot())
    test('Exist title', () => expect(mdTools.h5('Text')).toMatchSnapshot())
  })

  describe(mdTools.text.name, () => {
    test('No content', () => expect(mdTools.text()).toMatchSnapshot())
    test('Exist content', () => expect(mdTools.text('text')).toMatchSnapshot())
  })

  test(mdTools.anchor.name, () => expect(mdTools.anchor('key', 'https://example.com/pathname')).toMatchSnapshot())
  test(mdTools.hyperlink.name, () => expect(mdTools.hyperlink('label', 'https://example.com/pathname')).toMatchSnapshot())
  test(mdTools.hyperlinkWithKey.name, () => expect(mdTools.hyperlinkWithKey('label', 'key')).toMatchSnapshot())
  describe(mdTools.image.name, () => {
    test('No alt', () => expect(mdTools.image('https://example.com/pathname')).toMatchSnapshot())
    test('Exist alt', () => expect(mdTools.image('https://example.com/pathname', 'alt')).toMatchSnapshot())
  })
  describe(mdTools.imageByKey.name, () => {
    test('No alt', () => expect(mdTools.imageByKey('key')).toMatchSnapshot())
    test('Exist alt', () => expect(mdTools.imageByKey('key', 'alt')).toMatchSnapshot())
  })
  describe(mdTools.listItem.name, () => {
    test('No content', () => expect(mdTools.listItem('')).toMatchSnapshot())
    test('Exist content', () => expect(mdTools.listItem('Text')).toMatchSnapshot())
  })
  describe(mdTools.taskItem.name, () => {
    test('No content', () => expect(mdTools.taskItem('')).toMatchSnapshot())
    test('Exist content', () => expect(mdTools.taskItem('Text')).toMatchSnapshot())
    test('No content with empty options', () => expect(mdTools.taskItem('', { })).toMatchSnapshot())
    test('Exist content with empty options', () => expect(mdTools.taskItem('Text', { })).toMatchSnapshot())
    test('No content and unselect', () => expect(mdTools.taskItem('', { selected: false })).toMatchSnapshot())
    test('Exist content and unselect', () => expect(mdTools.taskItem('Text', { selected: false })).toMatchSnapshot())
    test('No content and selected', () => expect(mdTools.taskItem('', { selected: true })).toMatchSnapshot())
    test('Exist content and selected', () => expect(mdTools.taskItem('Text', { selected: true })).toMatchSnapshot())
  })
})
