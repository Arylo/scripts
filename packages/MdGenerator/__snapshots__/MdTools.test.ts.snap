// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Markdown Tools > anchor 1`] = `"[key]: https://example.com/pathname"`;

exports[`Markdown Tools > enter 1`] = `
"
"
`;

exports[`Markdown Tools > h1 > Exist title 1`] = `"# Text"`;

exports[`Markdown Tools > h1 > No title 1`] = `"#"`;

exports[`Markdown Tools > h2 > Exist title 1`] = `"## Text"`;

exports[`Markdown Tools > h2 > No title 1`] = `"##"`;

exports[`Markdown Tools > h3 > Exist title 1`] = `"### Text"`;

exports[`Markdown Tools > h3 > No title 1`] = `"###"`;

exports[`Markdown Tools > h4 > Exist title 1`] = `"#### Text"`;

exports[`Markdown Tools > h4 > No title 1`] = `"####"`;

exports[`Markdown Tools > h5 > Exist title 1`] = `"##### Text"`;

exports[`Markdown Tools > h5 > No title 1`] = `"#####"`;

exports[`Markdown Tools > hyperlink 1`] = `"[label](https://example.com/pathname)"`;

exports[`Markdown Tools > hyperlinkWithKey 1`] = `"[label][key]"`;

exports[`Markdown Tools > image > Exist alt 1`] = `"![alt](https://example.com/pathname)"`;

exports[`Markdown Tools > image > No alt 1`] = `"![](https://example.com/pathname)"`;

exports[`Markdown Tools > imageByKey > Exist alt 1`] = `"![alt][key]"`;

exports[`Markdown Tools > imageByKey > No alt 1`] = `"![][key]"`;

exports[`Markdown Tools > indent > Level 1 1`] = `""`;

exports[`Markdown Tools > indent > Level 1 with text 1`] = `"text"`;

exports[`Markdown Tools > indent > Level 2 1`] = `"  "`;

exports[`Markdown Tools > indent > Level 2 with text 1`] = `"  text"`;

exports[`Markdown Tools > indent > Level 3 1`] = `"    "`;

exports[`Markdown Tools > indent > Level 3 with text 1`] = `"    text"`;

exports[`Markdown Tools > indent > Level 4 1`] = `"      "`;

exports[`Markdown Tools > indent > Level 4 with text 1`] = `"      text"`;

exports[`Markdown Tools > indent > Level 5 1`] = `"        "`;

exports[`Markdown Tools > indent > Level 5 with text 1`] = `"        text"`;

exports[`Markdown Tools > listItem > Exist content 1`] = `"- Text"`;

exports[`Markdown Tools > listItem > No content 1`] = `"-"`;

exports[`Markdown Tools > taskItem > Exist content 1`] = `"[ ] Text"`;

exports[`Markdown Tools > taskItem > Exist content and selected 1`] = `"[x] Text"`;

exports[`Markdown Tools > taskItem > Exist content and unselect 1`] = `"[ ] Text"`;

exports[`Markdown Tools > taskItem > Exist content with empty options 1`] = `"[ ] Text"`;

exports[`Markdown Tools > taskItem > No content 1`] = `"[ ]"`;

exports[`Markdown Tools > taskItem > No content and selected 1`] = `"[x]"`;

exports[`Markdown Tools > taskItem > No content and unselect 1`] = `"[ ]"`;

exports[`Markdown Tools > taskItem > No content with empty options 1`] = `"[ ]"`;

exports[`Markdown Tools > text > Exist content 1`] = `"text"`;

exports[`Markdown Tools > text > No content 1`] = `""`;
